### ISSUE TITLE
Access to KeePassium iCloud Drive folder

### ISSUE THREAD

**Issue by decodism** [keepassium/KeePassium#208]
Hello,

I don't think this is currently implemented. It would be nice to have access to the KeePassium folders in the Files app, either on the device or iCloud Drive, to be able to store KDBX files there.

Edit: I had not noticed before but it seems that it is already possible on the device.

**Comment by keepassium** [comment]
Hi! 

Yes, KeePassium already presents its folder in the Files app (pretty much from day 1). Normally, it should be visible immediately after installation.

However, sometimes the system chooses to hide the app folders which are empty. I have not found any pattern, but this [happens](https://stackoverflow.com/questions/60745368/cant-see-my-app-folder-under-ios-files-app-on-ipad-but-only-on-iphone) with other apps, too.

There are three possible workarounds:

- Import any key file to KeePassium. (Open the Files app → long-press any non-DB file → Share → KeePassium.)
- Copy any file to KeePassium's folder via [Finder/iTunes file sharing](https://support.apple.com/HT201301)
- Use the Files app to move any file to KeePassium, as described [here](https://stackoverflow.com/a/60813904/1671985).

Once the app folder is no longer empty, the system will show it in the Files. (As it should have done in the first place.)

**Comment by decodism** [comment]
What about accessing the KeePassium folder on iCloud Drive? I know that KDBX files can be stored in any folder, but since I only use KeePassium, it seems to make sense to want to keep them in that folder.

**Comment by keepassium** [comment]
KeePassium does not explicitly work with iCloud, so I am not aware about KeePassium folder on iCloud Drive… Do you see one?

**Comment by decodism** [comment]
No, it was more of a feature request. I do not know the technical details but I know that some apps have an iCloud Drive folder accessible from the Files app and I wish KeePassium did.

**Comment by keepassium** [comment]
> No, it was more of a feature request. I do not know the technical details but I know that some apps have an iCloud Drive folder accessible from the Files app and I wish KeePassium did.

I struggle to see the point, though… You can create a folder manually, put your database there and point KeePassium to the database. What's the benefit of KeePassium cluttering user's storage with an auto-generated folder?

**Comment by decodism** [comment]
As you imply, not having to select the database and have it automatically associated. I do not know if an app can dynamically control the creation of this folder though.

**Comment by Allenxuxu** [comment]
> I have the same problem. I want to see this file in the iCloud Drive directory, otherwise I won't know if iCloud has safely completed the synchronization/backup for me.
> 
> Just set NSUbiquitousContainerIsDocumentScopePublic in Info.plist. I hope this switch can be adopted and turned on.
> 
> There is no need to load automatically when a new app is installed. It can still remain manual, but users should be able to see this file in iCloud Drive.

https://stackoverflow.com/questions/25203697/exposing-an-apps-ubiquitous-container-to-icloud-drive-in-ios-8

**Comment by keepassium** [comment]
> I have the same problem. I want to see this file in the iCloud Drive directory, otherwise I won't know if iCloud has safely completed the synchronization/backup for me.

@Allenxuxu , this was a [feature request](https://github.com/keepassium/KeePassium/issues/208#issuecomment-984800194), iCloud Drive is already accessible by the app, and automatic database creation/association was rejected [for these reasons](https://github.com/keepassium/KeePassium/issues/208#issuecomment-990218328).

Your issue is different, but still does not apply to KeePassium. You ask to expose the ubiquitous container which KeePassium does not use in the first place. All the databases referenced by the app are in publicly visible folders. (With the exception of [in-app backup](https://support.keepassium.com/docs/database-backup/) which is irrelevant here.) 

In summary: when you create a database in KeePassium, you can only save it to a public folder visible in iOS Files app and (directly related) system file picker dialog. There is no hidden location to be "made public". This is by design: users have access to their data even if the app fails to launch.

**Comment by layday** [comment]
I'm trying to make the switch from another password manager and I cannot for the life of me figure out how to view the KeePassium folder on iCloud Drive.  I'd like to use KeePassium on iOS and KeePassXC on macOS.  I created a database on my phone using KeePassium which was stored in the "KeePassium" folder on iCloud Drive.  I want to import this database into KeePassXC.   The folder and therefore the database is not visible on iCloud.  Is there some other way that this is all supposed to work?
